### PR TITLE
Certificate access control & Template design changes

### PR DESCRIPTION
**Pull request by arslanashraf7**
#### What are the relevant tickets?
This a part of https://github.com/mitodl/bootcamp-ecommerce/issues/1103

**NOTE:** This is based on Asad's PR for main certificate template changes. https://github.com/mitodl/bootcamp-ecommerce/pull/1116. This will be merged in base PR first.

#### What's this PR do?
It adds Below steps of the acceptance criteria in the mentioned ticker above.

- [x] **IMPORTANT** the user view should be behind a feature flag
- [x] when the bootcamper is logged in, and viewing their own certificate, include sharing UI (same as xPRO)
- [x] print CSS omits header and footer and **prints on one page horizontally**
- [x] return a 404 when the certificate has been revoked 

#### How should this be manually tested?
- Generate & enable certificates for any bootcamp through CMS
- View the certificate

#### Where should the reviewer start?
Configuring certificates for bootcamps through CMS

#### Screenshots (if appropriate)
Screenshot 2020-12-30 at 4 06 45 PM

**Print**
Screenshot 2020-12-30 at 4 07 04 PM

**Mobile**
Screenshot 2020-12-30 at 4 07 52 PM

### CODE REVIEW

File: static/scss/certificates/certificate.scss [hunk 1]
```diff
@@ -292,11 +437,34 @@
...
-    color: #333;
+    color: $cert-img-holder-color;
   }
 }
 
 @media print {
+  a {
+    color: black !important;
+  }
+
```
Comment (HamzaIbnFarooq) [inline static/scss/certificates/certificate.scss:448]: Kindly add the following to restrict `help` button from getting printed:
```
#launcher {
    display: none;
}
```

File: cms/templates/certificate_page.html [hunk 2]
```diff
@@ -1,57 +1,123 @@
...
+                            </a>
+                        </li>
+                        <li>
+                            <a href="javascript:window.print();">
+                                <img src="{% static 'images/certificates/icon-print.svg' %}" alt="Print">
+                            </a>
+                        </li>
+                    </ul>
+                        <h2>Congratulations, {{ learner_name }}!</h2>
+                        <p>You have successfully completed {{ page.product_name }}. Share your accomplishment with your friends, family and colleagues. </p>
```
Comment (HamzaIbnFarooq) [inline cms/templates/certificate_page.html:36]:
```suggestion
                        <p>You have successfully completed {{ page. bootcamp_run_name }}. Share your accomplishment with your friends, family and colleagues. </p>
```

**Review by HamzaIbnFarooq — APPROVED** [review]
LG, just some minor tweaks.

File: cms/templates/certificate_page.html (hunk 2 above)
Comment (arslanashraf7) [reply cms/templates/certificate_page.html]: Updated

File: static/scss/certificates/certificate.scss (hunk 1 above)
Comment (arslanashraf7) [reply static/scss/certificates/certificate.scss:448]: Updated

**Pull request merged**